### PR TITLE
Added support for webhook parsing

### PR DESCRIPTION
**Pull request by Samuelhertzberg**
Added a file and class "RequestParser.java"

The class contains the method 'parse' which is passed a HttpServletRequest which is parsed to a JSONObject. Then the appropriate fields are extracted.

This should resolve issue #8

### CODE REVIEW

**Review by apeinot — CHANGES_REQUESTED** [review]
Your function and test looks good. They resolve #8 
Can you also parse the following fields that will be usefull for commands:
- repository/clone_url
- repository/name

Add `package CIserver;` at the beginning of your main class file

**Review by francislaus — APPROVED** [review]

**Comment by francislaus** [comment]
Apart from what @apeinot requested, I think that the implementation and documentation of both the function itself and the test cases looks good.

**Review by Jonagoo — CHANGES_REQUESTED** [review]
Which method is supposed to be called from the main method? I assume it is parse, in which case I think a few more fields from the json is needed as getters, the ssh-url in particular for cloning as well as the name of the repository for more convenient running of tests.

Aside from this everything looks good.

**Comment by apeinot** [comment]
> Which method is supposed to be called from the main method? I assume it is parse, in which case I think a few more fields from the json is needed as getters, the ssh-url in particular for cloning as well as the name of the repository for more convenient running of tests.

I think the clone_url that I asked for is enough (we don't need the ssh_url to clone)

**Comment by Jonagoo** [comment]
> > Which method is supposed to be called from the main method? I assume it is parse, in which case I think a few more fields from the json is needed as getters, the ssh-url in particular for cloning as well as the name of the repository for more convenient running of tests.
> 
> I think the clone_url that I asked for is enough (we don't need the ssh_url to clone)

You're probably correct, I've just always went with ssh when cloning so that was the one I recognised.

**Comment by Samuelhertzberg** [comment]
Will fix asap

**Comment by Jonagoo** [comment]
Another good thing would be to give some sort of return code to see if parse() executed successfully. Just returning a 0 if it's successful and 1 otherwise or something similar would help prevent exceptions later in the code.

**Comment by Samuelhertzberg** [comment]
Changed committed and pushed. Error codes specified in parse() doc-comments.

**Review by Jonagoo — APPROVED** [review]

File: src/main/java/RequestParser.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+     String after = null;
+     String statuses_url = null;
+     String clone_url = null;
+     String name = null;
+     int statusCode;
+
+    /**
+    Parse request JSON and get the correct fields
+    @param request The http(s) POST request
+    @return The status code. 1 for success, 0 for connection/response issues and -1 for parsing issues.
```
Comment (emilelm96) [inline src/main/java/RequestParser.java:19]: The parse() function doesn't seem to return -1 when there are parsing issues like specified in the documentation.

**Review by emilelm96 — APPROVED** [review]

**Review by apeinot — APPROVED** [review]

**Pull request merged**